data_steward/cdr_cleaner/reporter.py

```diff
@@ -0,0 +1,306 @@
+    rules report.  Adds the documentation string of the function only if a
+    description is requested.
+
+    :param func:  The function that is part of the clean rules report.
+    :param fields_list: The list of fields the user requested.
+
+    :return: A dictionary of values representing the known and requested
+        fields for the given function.
+    """
+    func_info = dict()
+
+    for field in fields_list:
+        func_info[field] = 'unknown'
+
+    if 'name' not in fields_list:
+        LOGGER.info(f"Adding 'name' field to notify report reader this "
+                    f"function ({func.__qualname__}), "
+                    "needs to be implemented as a class.")
+
+    func_info['name'] = func.__name__
+
+    if 'module' not in fields_list:
+        LOGGER.info(f"Adding 'module' field to notify report reader this "
+                    f"function ({func.__qualname__}) "
+                    "needs to be implemented as a class.")
+
+    func_info['module'] = func.__module__
+
+    if 'description' in fields_list:
+        func_info['description'] = func.__doc__
+
+    return func_info
+
+
+def get_stage_elements(data_stage, fields_list):
+    """
+    Return the field info for rules defined for this data_stage.
+
+    For the given data_stage, this will determine the values of the
+    requested fields.  This information will be returned as a list
+    of dictionaries to preserve the information order.  Each dictionary
+    contains values for the report for a single cleaning rule.
+
+    :param data_stage: The data stage to report for.
+    :param fields_list: The user defined fields to report back.
+
+    :returns: a list of dictionaries representing the requested fields.
+    """
+
+    report_rows = []
+    for rule in control.DATA_STAGE_RULES_MAPPING.get(data_stage, []):
+        rule_info = {}
+
+        try:
+            clazz = rule[0]
+            instance = clazz('foo', 'bar', 'baz')
+            LOGGER.info(f"{clazz} ducktyped to a class")
+
+        except (RuntimeError, TypeError, HttpError, BadRequest):
+            LOGGER.info(f"{rule} did NOT ducktype to a class")
+            rule_info = get_function_info(clazz, fields_list)
+            report_rows.append(rule_info)
+
+        else:
+            try:
+                # this is a class
+                _ = instance.get_query_specs()
+                LOGGER.info(f"{clazz} is a class")
+                for field in fields_list:
+                    try:
+                        value = 'NO DATA'
+                        if field in report.FIELDS_PROPERTIES_MAP:
+                            func = report.FIELDS_PROPERTIES_MAP[field]
+                            value = getattr(instance, func, 'no data')
+                        elif field in report.FIELDS_METHODS_MAP:
+                            func = report.FIELDS_METHODS_MAP[field]
+                            value = getattr(instance, func, 'no data')()
```

Comment:
getting an error when passing in `sandbox-tables` as a field argument. It seems to be hitting the function `get_sandbox_table_names` and returns an error when trying to get the value. 

```
Traceback (most recent call last):
  File "/Applications/PyCharm CE.app/Contents/helpers/pydev/pydevd.py", line 2073, in <module>
    main()
  File "/Applications/PyCharm CE.app/Contents/helpers/pydev/pydevd.py", line 2067, in main
    globals = debugger.run(setup['file'], None, None, is_module)
  File "/Applications/PyCharm CE.app/Contents/helpers/pydev/pydevd.py", line 1418, in run
    return self._exec(is_module, entry_point_fn, module_name, file, globals, locals)
  File "/Applications/PyCharm CE.app/Contents/helpers/pydev/pydevd.py", line 1425, in _exec
    pydev_imports.execfile(file, globals, locals)  # execute the script
  File "/Applications/PyCharm CE.app/Contents/helpers/pydev/_pydev_imps/_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "/Users/aimeemcgrenera/Documents/dev/curation/data_steward/cdr_cleaner/reporter.py", line 306, in <module>
    main()
  File "/Users/aimeemcgrenera/Documents/dev/curation/data_steward/cdr_cleaner/reporter.py", line 299, in main
    write_csv_report(args.output_filepath, args.data_stage, args.fields)
  File "/Users/aimeemcgrenera/Documents/dev/curation/data_steward/cdr_cleaner/reporter.py", line 257, in write_csv_report
    required_fields_dict = get_stage_elements(stage, fields_list)
  File "/Users/aimeemcgrenera/Documents/dev/curation/data_steward/cdr_cleaner/reporter.py", line 118, in get_stage_elements
    value = getattr(instance, func, 'no data')()
  File "/Users/aimeemcgrenera/Documents/dev/curation/data_steward/cdr_cleaner/cleaning_rules/clean_mapping.py", line 229, in get_sandbox_tablenames
    raise NotImplementedError("Please fix me")
NotImplementedError: Please fix me```

Reply:
DC-974 created.